### PR TITLE
YLP-2671 Group alarms with the same code occurring in a 30 mn interval

### PR DESCRIPTION
**Pull request by celia-rouis**
## Checklist
- [x] PR name contains the Jira ticket number
- [x] Changelog updated
- [x] Code tested (unit tests and integration tests)
- [x] No console errors/warnings in integration tests
- [x] No translation files updated (except for English files)
- [x] Code formatted on every file
- [x] Copyright added on every file
- [x] End-to-end tests successful on CI

### CODE REVIEW

**Review by TimCanu — APPROVED** [review]

File: packages/medical-domain/src/index.ts [hunk 1]
```diff
@@ -154,6 +155,7 @@ export {
   type PumpSettingsParameter,
   ChangeType,
   BasalBolusStatisticsService,
-  MS_IN_DAY
+  MS_IN_DAY,
```
[the remark this answers is not in the document]
Comment (celia-rouis) [reply packages/medical-domain/src/index.ts:158]: Yes I agree 😄 Maybe we can plan a discussion around this topic and see the next steps to get rid of these barrel files?

File: packages/medical-domain/src/domains/repositories/medical/datum/alarm-event.service.ts [hunk 2]
```diff
@@ -121,10 +192,11 @@ const filterOnDate = (data: AlarmEvent[], start: number, end: number, weekDaysFi
   return DatumService.filterOnDate(data, start, end, weekDaysFilter) as AlarmEvent[]
 }
 
-const AlarmEventService: DatumProcessor<AlarmEvent> = {
+const AlarmEventService: DatumProcessor<AlarmEvent> & AlarmEventProcessor = {
```
[the remark this answers is not in the document]
Comment (celia-rouis) [reply packages/medical-domain/src/domains/repositories/medical/datum/alarm-event.service.ts:195]: I created a dedicated Jira ticket to deal with this issue: https://diabeloop.atlassian.net/browse/YLP-2679
I could even start working on it right after this PR!

File: packages/medical-domain/src/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (TimCanu) [reply packages/medical-domain/src/index.ts:158]: The best thing would be for someone to see if it really has an impact... So if one of us is bored one day... They can investigate :wink:

**Pull request merged**